staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -931,6 +943,8 @@ func (r *Request) request(ctx context.Context, fn func(*http.Request, *http.Resp
 
 			retries++
 			if seconds, wait := checkWait(resp); wait && retries <= r.maxRetries {
```

Comment:
do we have a client-side 429 metric?  I know we have some metrics

Reply:
Yes, `client-go` provides sink interfaces for metrics. `component-base` implementation does provide a metric `rest_client_requests_total` which has a label called `code`. It should have `429` if the server has rejected the request. 

we also have a metric `rest_client_rate_limiter_duration_seconds` that tracks the client-side rate limiter latency by verb and url.
Not all k8s application/controller use the default `component-base` implementation though.

Reply:
> Yes, `client-go` provides sink interfaces for metrics. `component-base` implementation does provide a metric `rest_client_requests_total` which has a label called `code`. It should have `429` if the server has rejected the request.

openshift largely does. Gather those numbers on your dashboard so we can slice by client binary.